src/ripple/app/misc/Transaction.h

```diff
@@ -171,6 +172,21 @@ class Transaction
     beast::Journal  j_;
 };
 
+struct TransactionEnvelope
+{
+public:
+    TransactionEnvelope (std::shared_ptr<Transaction> txn,
+                         std::string const& completeLedgers)
+        : txn (txn)
+        , completeLedgers (completeLedgers)
+    {}
+
+    TransactionEnvelope () = default;
```

Comment:
I probably wouldn't make this default constructable.

Reply:
Removed.